api/admin/controller/work_editor.py

```diff
@@ -961,7 +961,9 @@ def custom_lists(self, identifier_type, identifier):
                         affected_lanes.add(lane)
 
             # If any list changes affected lanes, update their sizes.
+            # NOTE: This may not make a difference until the
+            # works are actually re-indexed.
             for lane in affected_lanes:
-                lane.update_size(self._db)
+                lane.update_size(self._db, self.search_engine)
```

Comment:
I don't think this was working before. For lane size to be affected by a work being added to or removed from a list, the materialized view would have to be updated.  The mechanism has changed from "materialized view" to "search index", but the data that would actually change the count still hasn't been propagated to that mechanism. It's something that happens in the background.

(This isn't the case for when a list is added to or removed from a lane, so the rest of this branch isn't affected.)

I didn't remove the code because I'm not sure I'm right and this is in the middle of a much bigger change.

Reply:
It's not a huge deal for this because the page where you add and remove lists from an individual work does not show lane sizes itself. But in general it is important for the admin interface to have a way to get updated data.

Reply:
Long-term we'll be in a better position, because the search index is updated much more frequently than the materialized view. For example, we can flag a lane as needing a size refresh, and have a script that runs once an hour recalculating the size of any affected lanes.

Reply:
Once an hour is still not enough for this use case though. If you make a change in the admin interface, you expect it to display immediately on the page you're looking at.

Reply:
I don't think we disagree.

If you remove a work from a custom list and need it to immediately be reflected in `lane.size` then you need to reindex the work and then call Lane.update_size(). That's the bit that I don't think was working before. You can force one work to be reindexed right now, but you can't force an update to the materialized view.

If you remove a custom list from a lane, then calling Lane.update_size() (which happens now) is enough to update the size. The list of custom lists goes into the `Filter`, so it takes effect immediately. The presence of a work on a custom list goes into the search index, so the work has to be reindexed before Lane.update_size() will notice.